js_modules/dagster-ui/packages/ui-core/src/asset-data/AssetLiveDataThread.tsx

```diff
@@ -0,0 +1,255 @@
+      const id = tokenForAssetKey(assetNode.assetKey);
+      nodesByKey[id] = assetNode;
+    });
+    data.assetsLatestInfo.forEach((assetLatestInfo) => {
+      const id = tokenForAssetKey(assetLatestInfo.assetKey);
+      liveDataByKey[id] = buildLiveDataForNode(nodesByKey[id]!, assetLatestInfo);
+    });
+
+    this.manager._updateCache(liveDataByKey);
+    return liveDataByKey;
+  }
+
+  private async _batchedQueryAssets(assetKeys: AssetKeyInput[]) {
+    if (!assetKeys.length || this.isFetching) {
+      return;
+    }
+    this.isFetching = true;
+    this.manager._markAssetsRequested(assetKeys);
+
+    const doNextFetch = () => {
+      this.isFetching = false;
+      const nextAssets = this.manager.determineAssetsToFetch(this.getObservedKeys());
+      if (nextAssets.length) {
+        this._batchedQueryAssets(nextAssets);
+      }
+    };
+    try {
+      const data = await this.queryAssetKeys(assetKeys);
+      this.manager._updateFetchedAssets(assetKeys, data);
+      doNextFetch();
+    } catch (e) {
+      console.error(e);
+
+      if ((e as any)?.message?.includes('500')) {
+        // Mark these assets as fetched so that we don't retry them until after the poll interval rather than retrying them immediately.
+        // This is preferable because if the assets failed to fetch it's likely due to a timeout due to the query being too expensive and retrying it
+        // will not make it more likely to succeed and it would add more load to the database.
+        this.manager._updateFetchedAssets(assetKeys, {});
+      } else {
+        // If it's not a timeout from the backend then lets keep retrying instead of moving on.
+        this.manager._unmarkAssetsRequested(assetKeys);
+      }
+
+      setTimeout(
+        () => {
+          doNextFetch();
+        },
+        // If the poll rate is faster than 5 seconds lets use that instead
+        Math.min(this.pollRate, 5000),
+      );
+    }
+  }
+}
+
+export const ASSET_LATEST_INFO_FRAGMENT = gql`
+  fragment AssetLatestInfoFragment on AssetLatestInfo {
+    id
+    assetKey {
+      path
+    }
+    unstartedRunIds
+    inProgressRunIds
+    latestRun {
+      id
+      ...AssetLatestInfoRun
+    }
+  }
+
+  fragment AssetLatestInfoRun on Run {
+    status
+    endTime
+    id
+  }
+`;
+
+export const ASSET_NODE_LIVE_FRAGMENT = gql`
+  fragment AssetNodeLiveFragment on AssetNode {
+    id
+    opNames
+    repository {
+      id
+    }
+    assetKey {
+      path
+    }
+    assetMaterializations(limit: 1) {
+      ...AssetNodeLiveMaterialization
+    }
+    assetObservations(limit: 1) {
+      ...AssetNodeLiveObservation
+    }
+    assetChecksOrError {
+      ... on AssetChecks {
+        checks {
+          ...AssetCheckLiveFragment
+        }
+      }
+    }
+    freshnessInfo {
+      ...AssetNodeLiveFreshnessInfo
+    }
+    staleStatus
+    staleCauses {
+      key {
+        path
+      }
+      reason
+      category
+      dependency {
+        path
+      }
+    }
```

Comment:
I'll look into this in a follow-up, I think to do it I'd like to refactor this AssetLiveDataProvider even more so that we can do chunking/batching for a variety of queries instead of hardcoding just this one.